domain/fhir/types.go

```diff
@@ -55,3 +55,18 @@ type Composition struct {
 	Title      datatypes.String          `json:"title,omitempty"`
 	Section    []CompositionSection      `json:"section,omitempty"`
 }
+
+// EpisodeOfCare defines a basic FHIR STU3 EpisodeOfCare resource which is currently not included in the FHIR library.
+type EpisodeOfCare struct {
+	resources.Base
+	Identifier           datatypes.Identifier        `json:"identifier"`
+	Status               datatypes.Code              `json:"status"`
+	Type                 []datatypes.CodeableConcept `json:"type"`
+	Patient              datatypes.Reference         `json:"patient"`
+	ManagingOrganization datatypes.Reference         `json:"managingOrganization"`
+	Period               datatypes.Period            `json:"period"`
+	ReferralRequest      datatypes.Reference         `json:"referralRequest"`
+	CareManager          datatypes.Reference         `json:"careManager"`
+	Team                 datatypes.Reference         `json:"team"`
+	Account              datatypes.Reference         `json:"account"`
```

Comment:
I think they need a omitempty hint?

Reply:
Why? `datatypes.Reference` isn't a pointer so the key/value won't ever be omitted (as it can't be `nil`). Example: https://play.golang.org/p/ogH77sPGns4